app/service/game_service.py

```diff
@@ -88,13 +98,16 @@ def _assert_game_exists(game_id: str) -> None:
 
 
 @trace("game_service")
-def assert_valid_game_member(game_id: str, member_id: str, user_id: str) -> None:
-    _assert_game_exists(game_id)
+def _assert_valid_game_member(game_id: str, member_id: str, user_id: str) -> None:
```

Comment:
Two issues with this:
1. Type annotations is wrong this currently returns Tuple[Game, GameMember]
2. This should probably not explicitly assert the game existence. It is kind of concussing that _assert_valid_game_member returns both a Game and a GameMember. Better if it only returns a GameMember and the _assert_game_exists is done outside this function. Probably between line 83 and 84.
Besides If the game does not exist it is quite unlikely that member_repo.find(member_id) would return anything

So in conclusion I would move line 102 to 84 and only return GameMember from this function

Reply:
Updated but need second review. Moved _assert_game_exist to row 84 and _assert_valid_game_member only asserts if a member matches user id. 

Also updated _assert_valid_game_member to check that the member is part of the correct game, else throws 428. Else game_id is not doing anything in routes for "leave game" and "set user ready".